src/Nerdbank.Streams/MultiplexingStream.cs

```diff
@@ -1113,6 +1180,62 @@ private void OnChannelDisposed(Channel channel)
             }
         }
 
+        /// <summary>
+        /// Called when the local end was not able to completely write all the data to this channel due to an error,
+        /// leading to the transmission of a <see cref="ControlCode.ContentWritingError"/> frame being sent for this channel.
+        /// </summary>
+        /// <param name="channel">The channel whose writing was halted.</param>
+        /// <param name="exception">The exception that caused the writing to be haulted.</param>
+        private void OnChannelWritingError(Channel channel, Exception exception)
+        {
+            Requires.NotNull(channel, nameof(channel));
+
+            // Make sure that we are allowed to send error frames on this protocol version
+            if (this.protocolMajorVersion == 1)
+            {
+                if (this.TraceSource!.Switch.ShouldTrace(TraceEventType.Information))
+                {
+                    this.TraceSource.TraceEvent(
+                        TraceEventType.Information,
+                        (int)TraceEventId.WriteError,
+                        "Not informing remote side of write error on channel {0} since MultiplexingStream has protocol version of {1}",
+                        channel.QualifiedId,
+                        this.protocolMajorVersion);
+                }
+
+                return;
+            }
+
+            // Verify that we are able to communicate to the remote side on this channel
+            if (this.channelsPendingTermination.Contains(channel.QualifiedId) || !this.openChannels.ContainsKey(channel.QualifiedId))
+            {
+                if (this.TraceSource!.Switch.ShouldTrace(TraceEventType.Information))
```

Comment:
There is no known case because a channel will either close because it encountered this error during the sending of the data or that it finished writing all the data, in which case the method will be never be invoked. This is more of a sanity check than anything else.